apps/src/lab2/views/dialogs/GenericDialog.tsx

```diff
@@ -36,6 +36,7 @@ export type GenericDialogProps = GenericDialogTitleProps &
       [key in ButtonType]?: {
         text?: string;
         callback?: dialogCallback;
+        disabled?: boolean;
```

Comment:
all buttons can now optionally be disabled.

e.g., This is handy for the `GenericPrompt` dialog which disables the `confirm` button unless a valid input is entered.